prez/sparql/methods.py

```diff
@@ -1,223 +1,132 @@
+    response: httpxResponse = client.post(
+        settings.sparql_endpoint,
+        data=query,
+        headers={
+            "Accept": "application/json",
+            "Content-Type": "application/sparql-query",
+        },
+        timeout=TIMEOUT,
+    )
     if 200 <= response.status_code < 300:
-        return True, response.json()["boolean"]
+        return True, response.json()["results"]["bindings"]
     else:
         return False, {
             "code": response.status_code,
             "message": response.text,
-            "prez": prez,
         }
 
 
-def sparql_ask_non_async(query: str, prez: str):
-    """Returns an rdflib Graph from a CONSTRUCT query for a single SPARQL endpoint"""
-    with Client() as client:
-        response: httpxResponse = client.post(
-            settings.sparql_creds[prez]["endpoint"],
-            data={"query": query},
-            headers={
-                "Accept": "*/*",
-                "Content-Type": "application/x-www-form-urlencoded",
-                "Accept-Encoding": "gzip, deflate",
-            },
-            timeout=TIMEOUT,
-        )
+def sparql_ask_non_async(query: str):
+    """Returns True if the provided triple pattern exists in the graph, otherwise False"""
+    response: httpxResponse = client.post(
+        settings.sparql_endpoint,
+        data={"query": query},
+        headers={
+            "Accept": "*/*",
+            "Content-Type": "application/x-www-form-urlencoded",
+            "Accept-Encoding": "gzip, deflate",
+        },
+        timeout=TIMEOUT,
+    )
     if 200 <= response.status_code < 300:
         return True, response.json()["boolean"]
     else:
         return False, {
             "code": response.status_code,
             "message": response.text,
-            "prez": prez,
         }
 
 
-def sparql_construct_non_async(query: str, prez: str):
-    """Returns an rdflib Graph from a CONSTRUCT query for a single SPARQL endpoint"""
-
-    with Client() as client:
-        response: httpxResponse = client.post(
-            settings.sparql_creds[prez]["endpoint"],
-            data=query,
-            headers={
-                "Accept": "text/turtle",
-                "Content-Type": "application/sparql-query",
-            },
-            timeout=TIMEOUT,
-        )
-    if 200 <= response.status_code < 300:
-        return True, Graph(bind_namespaces="rdflib").parse(data=response.text)
-    else:
-        return False, {
-            "code": response.status_code,
-            "message": response.text,
-            "prez": prez,
-        }
+async def sparql(request: Request):
+    """Sends a starlette Request object (containing a SPARQL query in the URL parameters) to a proxied SPARQL
+    endpoint."""
+    url = httpx.URL(query=request.url.query.encode("utf-8"))
+    rp_req = async_client.build_request(
+        request.method, url, headers=request.headers.raw, content=request.stream()
```

Comment:
Passing on the `Host` header here is problematic when sending requests to servers relying on virtual hosts reading the `Host` header value and matching routes on it.

Can we do a filter and either drop the `Host` header here or drop it and add the base URL of the SPARQL endpoint? Adding the base URL in the absence of a `Host` header I think is the default behaviour when it comes to machine tools such as Postman and curl.

Reply:
When testing with the GSQ Prez Fuseki via the Prez UI SPARQL page, I was getting a 404 response when sending queries that were proxied through Prez due to the `Host` header.

Reply:
This is what I've just pushed:

```async def sparql(request: Request):
    """Sends a starlette Request object (containing a SPARQL query in the URL parameters) to a proxied SPARQL
    endpoint."""
    url = httpx.URL(
        url=settings.sparql_endpoint, query=request.url.query.encode("utf-8")
    )
    rp_req = async_client.build_request(
        request.method, url, headers=request.headers.raw, content=request.stream()
    )
    return await async_client.send(rp_req, stream=True)```

So would it just be dropping the Host header in addition to this?

Reply:
I think something like this will work.

```diff
async def sparql(request: Request):
    """Sends a starlette Request object (containing a SPARQL query in the URL parameters) to a proxied SPARQL
    endpoint."""
    url = httpx.URL(url=settings.sparql_endpoint, query=request.url.query.encode("utf-8"))
    
+    headers = []
+    for header in request.headers.raw:
+        if header[0] != b'host':
+            headers.append(header)
+    headers.append((b'host', str(url.host).encode("utf-8")))
    
    rp_req = async_client.build_request(
-        request.method, url, headers=request.headers.raw, content=request.stream()
+        request.method, url, headers=headers, content=request.stream()
    )
    return await async_client.send(rp_req, stream=True)
```

Reply:
thanks for the code - added